src/Illuminate/Validation/Validator.php

```diff
@@ -2262,6 +2279,64 @@ protected function normalizeRule($rule)
     }
 
     /**
+     * Determine if the given rule accepts other field names as parameters.
+     * 
+     * @param  string  $rule
+     * @return bool
+     */
+    protected function acceptsFields($rule)
+    {
+        return in_array($rule, $this->fieldRules);
+    }
+
+    /**
+     * Get the numeric keys from an attribute flattened with dot notation.
+     *
+     * E.g. 'foo.1.bar.2.baz' -> [1, 2]
+     *
+     * @param  string  $attribute
+     * @return array
+     */
+    protected function getNumericKeys($attribute)
+    {
+        if (preg_match_all('/\.(\d+)\./', $attribute, $keys)) {
+            array_shift($keys);
+
+            return Arr::collapse($keys);
+        }
+
+        return [];
+    }
```

Comment:
You can simplify as so:
```
if (preg_match_all('/\.(\d+)\./', $attribute, $keys)) {
    return $keys[1];
}

return [];
```
In fact, `$keys[1]` is set even if there is no match, so you could even do:
```
preg_match_all('/\.(\d+)\./', $attribute, $keys);

return $keys[1];
```